src/Build/Collections/ItemDictionary.cs

```diff
@@ -464,31 +456,9 @@ public void Reset()
             /// </summary>
             public void Dispose()
             {
-                Dispose(true);
                 GC.SuppressFinalize(this);
             }
 
-            /// <summary>
-            /// The real disposer.
-            /// </summary>
-            private void Dispose(bool disposing)
-            {
-                if (disposing)
-                {
-                    if (_listEnumerator != null)
-                    {
-                        if (_itemEnumerator != null)
-                        {
-                            _itemEnumerator.Dispose();
```

Comment:
Don't we still need to dispose these owned resources?

Reply:
I wasn't sure, so I have added it back.